### PR TITLE
Add support for privileged containers

### PR DESCRIPTION
**Pull request by rbradford**
Add support for privileged containers. This necessitated adding a mechanism to give tenants permissions so that is also implemented in this PR.

### CODE REVIEW

File: payloads/start.go [hunk 1]
```diff
@@ -201,6 +201,10 @@ type WorkloadRequirements struct {
 
 	// NetworkNode specifies that this workload must be scheduled on a network node
 	NetworkNode bool `yaml:"network_node,omitempty"`
+
+	// Privileged indicates that this container workload should be run with increased
+	// permissionsl
```
Comment (rbradford) [inline payloads/start.go]: nit: typo. Looks like I mashed "l" rather than "."

File: ciao-controller/internal/datastore/sqlite3db_test.go [hunk 2]
```diff
@@ -1221,13 +1221,61 @@ func TestSQLiteDBUpdateTenant(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(ts) != 1 {
+		t.Fatalf("Unexpected number of tenants")
+	}
+
+	if !ts[0].Permissions.PrivilegedContainers {
+		t.Fatal("Expected tenant permission set correctly")
```
Comment (markdryan) [inline ciao-controller/internal/datastore/sqlite3db_test.go]: Does it matter if we don't delete the tenant or disconnect from the db on failure?

File: ciao-launcher/docker_test.go [hunk 3]
```diff
@@ -662,6 +662,78 @@ func TestDockerCreateImageWithResources(t *testing.T) {
...
+			VnicIP:     testutil.AgentIP,
+			Privileged: true,
+		}}
+
+	if err := d.createImage("bridge", "172.16.0.1", nil, nil); err != nil {
+		t.Fatalf("Unable to create image : %v", err)
+	}
+
+	if tc.hostConfig.Privileged != true {
+		t.Fatal("Expected privileged host config")
```
Comment (markdryan) [inline ciao-launcher/docker_test.go]: Not sure that it really matters, but these should probably be t.Error so deleteImage gets called.  deleteImage doesn't really do anything but still.

File: ciao-launcher/docker_test.go [hunk 4]
```diff
@@ -662,6 +662,78 @@ func TestDockerCreateImageWithResources(t *testing.T) {
...
+			break
+		}
+	}
+
+	if !found {
+		t.Fatal("Require security option not found")
+	}
+
+	count := 0
+	for _, b := range tc.hostConfig.Binds {
```
Comment (markdryan) [inline ciao-launcher/docker_test.go]: I'm not sure this tests the correct thing.  The test would pass if tc.hostConfig.Bind = []string{"/dev:/dev","/dev:/dev"}.

File: ciao-controller/internal/datastore/sqlite3db_test.go (hunk 2 above)
Comment (rbradford) [reply ciao-controller/internal/datastore/sqlite3db_test.go]: No not really. I spotted this too. Each test is run in it's own temporary in-memory DB, everything is thrown out on test exit.

File: ciao-launcher/docker_test.go (hunk 3 above)
Comment (rbradford) [reply ciao-launcher/docker_test.go]: Fixed.

File: ciao-launcher/docker_test.go (hunk 4 above)
Comment (rbradford) [reply ciao-launcher/docker_test.go]: Fixed.

**Comment by markdryan** [comment]
We could also do with a few words about this on the web site, maybe in Working with Workloads:

https://ciao-project.github.io/workloads.html

**Pull request merged**